mlx_graphs/data/data.py

```diff
@@ -0,0 +1,61 @@
+from typing import Optional
+
+import mlx.core as mx
+import mlx.data as dx
+
+
+class Data:
+    """
+    Represents a graph data object with optional features and labels.
+
+    Args:
+        edge_index (mlx.core.array, optional): edge index representing the topology of
+            the graph, with shape [2, num_edges].
+        node_features (mlx.core.array, optional): Array of shape [num_nodes, num_node_features]
+            containing the features of each node.
+        edge_features (mlx.core.array, optional): Array of shape [num_edges, num_edge_features]
+            containing the features of each edge.
+        graph_features (mlx.core.array, optional): Array of shape [num_graph_features]
+            containing the global features of the graph.
+        node_labels (mlx.core.array, optional): Array of shape [num_nodes, num_node_labels]
+            containing the labels of each node.
+        edge_labels (mlx.core.array, optional): Array of shape [num_edges, num_edge_labels]
+            containing the labels of each edge.
+        graph_labels (mlx.core.array, optional): Array of shape [num_graph_labels]
+            containing the global labels of the graph.
+        **kwargs: Additional keyword arguments to store any other custom attributes.
+    """
+
+    def __init__(
+        self,
+        edge_index: Optional[mx.array] = None,
+        node_features: Optional[mx.array] = None,
+        edge_features: Optional[mx.array] = None,
+        graph_features: Optional[mx.array] = None,
+        node_labels: Optional[mx.array] = None,
```

Comment:
Is it really necessary to have labels for all of them?
I don't see a use case where we want to do classification of node/edges at the same time.

Reply:
I think it makes sense to have them there as they're all optional. At least to have a base default structure.